specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkSecurityPerimeter.json

```diff
@@ -788,6 +940,165 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/networkSecurityPerimeters/{networkSecurityPerimeterName}/profiles/{networkSecurityPerimeterProfile}/checkMembers": {
+      "parameters": [
+        {
+          "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+        },
+        {
+          "$ref": "#/parameters/ResourceGroupNameParameter"
+        },
+        {
+          "$ref": "#/parameters/NetworkSecurityPerimeterNameParameter"
+        },
+        {
+          "$ref": "./network.json#/parameters/ApiVersionParameter"
+        },
+        {
+          "$ref": "#/parameters/NetworkSecurityPerimeterProfileParameter"
+        }
+      ],
+      "post": {
+        "tags": [
+          "NetworkSecurityPerimetersCheckMembers"
+        ],
+        "operationId": "NetworkSecurityPerimeters_CheckMembers",
+        "description": "Action to check members associated to an NSP Profile",
+        "parameters": [
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CheckMembers"
+            },
+            "description": "Parameter supplied to create the network security perimeter check members."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Check members, nothing is getting created or updated.",
+            "schema": {
+              "$ref": "#/definitions/CheckMembers"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Post Check Members": {
+            "$ref": "./examples/NSPCheckMembersPost.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Network/locations/{location}/queryNetworkSecurityPerimeter": {
+      "parameters": [
+        {
+          "$ref": "#/parameters/LocationParameter"
+        }
+      ],
+      "post": {
+        "tags": [
+          "QueryNetworkSecurityPerimeters"
+        ],
+        "operationId": "NetworkSecurityPerimeters_Query",
+        "description": "Action to query network security parameter",
+        "parameters": [
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/QueryNSPReqRes"
+            },
+            "description": "Parameter supplied to query network security parameter."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Query NSP using parameter guid, no resource created or updated.",
+            "schema": {
+              "$ref": "#/definitions/QueryNSPReqRes"
```

Comment:
>QueryNSPReqRes [](http://example.com/codeflow?start=37&length=14)

same question here

Reply:
The response contains similar data with a few more added keys so we are reusing the object.

Spec: https://msazure.visualstudio.com/One/_git/Networking-MNC?path=/doc/Design-Docs/Network-Security-Perimeter-Design-Specification.md&version=GBmaster&_a=preview&anchor=internal-api-to-get-perimeter-resource-id-from-perimeter-guid-on-nsp-rp

Reply:
POST are for actions, so my previous comment applies here as well.

Reply:
We need to send multiple different items in an array so we are using POST. This is an internal API.

Please refer: https://stackoverflow.microsoft.com/questions/258519